float32[] names
float32[] x_points
float32[] y_points
